### DISCUSSION TITLE
ERROR    Server with name ds920 is not defined

### DISCUSSION THREAD

**Discussion by valkhyr1** [Taxel/PlexTraktSync#1850] [Support] [answered]
When I run plextraktsync sync for the time I get the message

You are now logged into Trakt. Your Trakt credentials have been added in
.env and .pytrakt.json files.
ERROR    Server with name ds920 is not defined
         ╭───────────────── Traceback (most recent call last) ─────────────────╮
         │ /root/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/ │
         │ plextraktsync/config/ServerConfigFactory.py:19 in get_server        │
         │                                                                     │
         │   16 │   def get_server(self, name: str):                           │
         │   17 │   │   self.load()                                            │
         │   18 │   │   try:                                                   │
         │ ❱ 19 │   │   │   return PlexServerConfig(name=name, **self.servers[ │
         │   20 │   │   except KeyError:                                       │
         │   21 │   │   │   raise RuntimeError(f"Server with name {name} is no │
         │   22                                                                │
         ╰─────────────────────────────────────────────────────────────────────╯
         KeyError: 'ds920'

         During handling of the above exception, another exception occurred:

         ╭───────────────── Traceback (most recent call last) ─────────────────╮
         │ /root/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/ │
         │ plextraktsync/cli.py:26 in wrap                                     │
         │                                                                     │
         │    23 │   │   │   cmd = getattr(module, name)                       │
         │    24 │   │   │                                                     │
         │    25 │   │   │   try:                                              │
         │ ❱  26 │   │   │   │   cmd(*args, **kwargs)                          │
         │    27 │   │   │   except EOFError as e:                             │
         │    28 │   │   │   │   raise ClickException(f"Program requested term │
         │       connected: {e}")                                              │
         │    29 │   │   │   except ClickException as e:                       │
         │                                                                     │
         │ /root/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/ │
         │ plextraktsync/commands/sync.py:48 in sync                           │
         │                                                                     │
         │   45 │                                                              │
         │   46 │   ensure_login()                                             │
         │   47 │   wc = factory.walk_config.update(movies=movies, shows=shows │
         │ ❱ 48 │   w = factory.walker                                         │
         │   49 │                                                              │
         │   50 │   if ids:                                                    │
         │   51 │   │   for id in ids:                                         │
         │                                                                     │
         │ /usr/lib/python3.10/functools.py:981 in __get__                     │
         │                                                                     │
         │   978 │   │   │   │   # check if another thread filled cache while  │
         │   979 │   │   │   │   val = cache.get(self.attrname, _NOT_FOUND)    │
         │   980 │   │   │   │   if val is _NOT_FOUND:                         │
         │ ❱ 981 │   │   │   │   │   val = self.func(instance)                 │
         │   982 │   │   │   │   │   try:                                      │
         │   983 │   │   │   │   │   │   cache[self.attrname] = val            │
         │   984 │   │   │   │   │   except TypeError:                         │
         │                                                                     │
         │ /root/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/ │
         │ plextraktsync/util/Factory.py:185 in walker                         │
         │                                                                     │
         │   182 │   │   from plextraktsync.plan.Walker import Walker          │
         │   183 │   │                                                         │
         │   184 │   │   walk_config = self.walk_config                        │
         │ ❱ 185 │   │   plex = self.plex_api                                  │
         │   186 │   │   trakt = self.trakt_api                                │
         │   187 │   │   mf = self.media_factory                               │
         │   188 │   │   pb = self.progressbar                                 │
         │                                                                     │
         │ /usr/lib/python3.10/functools.py:981 in __get__                     │
         │                                                                     │
         │   978 │   │   │   │   # check if another thread filled cache while  │
         │   979 │   │   │   │   val = cache.get(self.attrname, _NOT_FOUND)    │
         │   980 │   │   │   │   if val is _NOT_FOUND:                         │
         │ ❱ 981 │   │   │   │   │   val = self.func(instance)                 │
         │   982 │   │   │   │   │   try:                                      │
         │   983 │   │   │   │   │   │   cache[self.attrname] = val            │
         │   984 │   │   │   │   │   except TypeError:                         │
         │                                                                     │
         │ /root/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/ │
         │ plextraktsync/util/Factory.py:48 in plex_api                        │
         │                                                                     │
         │    45 │   │   from plextraktsync.plex.PlexApi import PlexApi        │
         │    46 │   │                                                         │
         │    47 │   │   return PlexApi(                                       │
         │ ❱  48 │   │   │   server=self.plex_server,                          │
         │    49 │   │   │   config=self.server_config,                        │
         │    50 │   │   )                                                     │
         │    51                                                               │
         │                                                                     │
         │ /usr/lib/python3.10/functools.py:981 in __get__                     │
         │                                                                     │
         │   978 │   │   │   │   # check if another thread filled cache while  │
         │   979 │   │   │   │   val = cache.get(self.attrname, _NOT_FOUND)    │
         │   980 │   │   │   │   if val is _NOT_FOUND:                         │
         │ ❱ 981 │   │   │   │   │   val = self.func(instance)                 │
         │   982 │   │   │   │   │   try:                                      │
         │   983 │   │   │   │   │   │   cache[self.attrname] = val            │
         │   984 │   │   │   │   │   except TypeError:                         │
         │                                                                     │
         │ /root/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/ │
         │ plextraktsync/util/Factory.py:76 in plex_server                     │
         │                                                                     │
         │    73 │   │   from plextraktsync.plex.PlexServerConnection import \ │
         │    74 │   │   │   PlexServerConnection                              │
         │    75 │   │                                                         │
         │ ❱  76 │   │   server = self.server_config                           │
         │    77 │   │                                                         │
         │    78 │   │   return PlexServerConnection(factory).connect(         │
         │    79 │   │   │   urls=server.urls,                                 │
         │                                                                     │
         │ /usr/lib/python3.10/functools.py:981 in __get__                     │
         │                                                                     │
         │   978 │   │   │   │   # check if another thread filled cache while  │
         │   979 │   │   │   │   val = cache.get(self.attrname, _NOT_FOUND)    │
         │   980 │   │   │   │   if val is _NOT_FOUND:                         │
         │ ❱ 981 │   │   │   │   │   val = self.func(instance)                 │
         │   982 │   │   │   │   │   try:                                      │
         │   983 │   │   │   │   │   │   cache[self.attrname] = val            │
         │   984 │   │   │   │   │   except TypeError:                         │
         │                                                                     │
         │ /root/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/ │
         │ plextraktsync/util/Factory.py:117 in server_config                  │
         │                                                                     │
         │   114 │   │   │   server_config.load()                              │
         │   115 │   │   │   server_name = config["PLEX_SERVER"]               │
         │   116 │   │                                                         │
         │ ❱ 117 │   │   return server_config.get_server(server_name)          │
         │   118 │                                                             │
         │   119 │   @cached_property                                          │
         │   120 │   def session(self):                                        │
         │                                                                     │
         │ /root/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/ │
         │ plextraktsync/config/ServerConfigFactory.py:21 in get_server        │
         │                                                                     │
         │   18 │   │   try:                                                   │
         │   19 │   │   │   return PlexServerConfig(name=name, **self.servers[ │
         │   20 │   │   except KeyError:                                       │
         │ ❱ 21 │   │   │   raise RuntimeError(f"Server with name {name} is no │
         │   22 │                                                              │
         │   23 │   def server_by_id(self, id: str):                           │
         │   24 │   │   self.load()                                            │
         ╰─────────────────────────────────────────────────────────────────────╯
         RuntimeError: Server with name ds920 is not defined
Error: Error running sync command: Server with name ds920 is not defined

How can I solve this?

**Comment by glensc** [comment, marked as answer]
did you read the welcome message of discussions? it tells you to report crashes as issues.

- https://github.com/Taxel/PlexTraktSync/discussions/803

**Comment by tundrablasen** [comment]
i have the exact same issue. how is that solved? @glensc answer was not very helpful

**Reply by glensc** [reply]
crashes need to be reported as issues. crashes reported as discussions will be ignored as they are incomplete.

**Comment by gurchgherkin** [comment]
Change the name of your plex server (within the plex settings) and then update the name in the servers.yml to match it.

Ensure your server name is simple, no punctuation, etc.

Run the terminal command again: docker compose run --rm plextraktsync sync

This seems to have worked for me, but I could not tell you why
